### PR TITLE
Build exceptions and handler

### PR DESCRIPTION
**Pull request by thomtrp**
Adding exceptions and handler for auth services.

Tested with:
- Workspace creation
- Workspace signup
- Workspace invitation
- Reset password
- Adding email account
- Impersonation

### CODE REVIEW

File: packages/twenty-server/src/engine/core-modules/auth/auth.resolver.ts [hunk 1]
```diff
@@ -61,154 +55,188 @@ export class AuthResolver {
...
-
-    assert(email, 'Invalid token', ForbiddenException);
+  async verify(@Args() verifyInput: VerifyInput): Promise<Verify | void> {
+    try {
+      const email = await this.tokenService.verifyLoginToken(
+        verifyInput.loginToken,
+      );
 
-    const result = await this.authService.verify(email);
+      const result = await this.authService.verify(email);
```
Comment (thomtrp) [inline packages/twenty-server/src/engine/core-modules/auth/auth.resolver.ts:173]: check on email has been put into this function

File: packages/twenty-server/src/engine/core-modules/auth/auth.resolver.ts [hunk 2]
```diff
@@ -217,53 +245,62 @@ export class AuthResolver {
...
-      args.passwordResetToken,
-    );
+  ): Promise<InvalidatePassword | void> {
+    try {
+      const { id } = await this.tokenService.validatePasswordResetToken(
+        args.passwordResetToken,
+      );
 
-    assert(id, 'User not found', NotFoundException);
+      await this.authService.updatePassword(id, args.newPassword);
```
Comment (thomtrp) [inline packages/twenty-server/src/engine/core-modules/auth/auth.resolver.ts:286]: check on user has been put into this function

File: packages/twenty-server/src/engine/core-modules/auth/auth.resolver.ts [hunk 3]
```diff
@@ -151,8 +137,6 @@ export class AuthResolver {
       verifyInput.loginToken,
     );
 
-    assert(email, 'Invalid token', ForbiddenException);
```
Comment (thomtrp) [inline packages/twenty-server/src/engine/core-modules/auth/auth.resolver.ts:154]: I moved the check to the function bellow

File: packages/twenty-server/src/engine/core-modules/auth/auth.resolver.ts [hunk 4]
```diff
@@ -188,10 +172,6 @@ export class AuthResolver {
 
   @Mutation(() => AuthTokens)
   async renewToken(@Args() args: AppTokenInput): Promise<AuthTokens> {
-    if (!args.appToken) {
-      throw new BadRequestException('Refresh token is mendatory');
-    }
-
```
Comment (thomtrp) [inline packages/twenty-server/src/engine/core-modules/auth/auth.resolver.ts:194]: same here

File: packages/twenty-server/src/engine/core-modules/auth/auth.resolver.ts [hunk 5]
```diff
@@ -205,10 +185,7 @@ export class AuthResolver {
     @Args() impersonateInput: ImpersonateInput,
     @AuthUser() user: User,
   ): Promise<Verify> {
-    // Check if user can impersonate
-    assert(user.canImpersonate, 'User cannot impersonate', ForbiddenException);
-
-    return this.authService.impersonate(impersonateInput.userId);
+    return await this.authService.impersonate(impersonateInput.userId, user);
```
Comment (thomtrp) [inline packages/twenty-server/src/engine/core-modules/auth/auth.resolver.ts:188]: same here

File: packages/twenty-server/src/engine/core-modules/auth/auth.resolver.ts [hunk 6]
```diff
@@ -240,20 +217,13 @@ export class AuthResolver {
...
-    );
-
-    assert(id, 'User not found', NotFoundException);
-
-    const { success } = await this.authService.updatePassword(
-      id,
-      args.newPassword,
-    );
+    const { id } =
+      await this.tokenService.validatePasswordResetToken(passwordResetToken);
```
Comment (thomtrp) [inline packages/twenty-server/src/engine/core-modules/auth/auth.resolver.ts:224]: same here

File: packages/twenty-server/src/engine/core-modules/auth/services/auth.service.ts [hunk 7]
```diff
@@ -70,15 +68,31 @@ export class AuthService {
...
+      throw new AuthException(
+        'User not found',
+        AuthExceptionCode.USER_NOT_FOUND,
+      );
+    }
+
+    if (!user.passwordHash) {
+      throw new AuthException(
+        'Incorrect login method',
+        AuthExceptionCode.FORBIDDEN_EXCEPTION,
```
[the remark this answers is not in the document]
Comment (thomtrp) [reply packages/twenty-server/src/engine/core-modules/auth/services/auth.service.ts:81]: Works for me!

**Review by Weiko — APPROVED** [review]
LGTM @thomtrp 👏

**Pull request merged**